site/content/deployment/openshift_s2i.md

```diff
@@ -0,0 +1,358 @@
+
+    oc new-project hono --display-name='Eclipse Hono™'
+
+Create the InfluxDB ConfigMap from the local file:
+
+    oc create configmap influxdb-config --from-file="../influxdb.conf"
+
+Then process and execute the main Hono template:
+
+    oc process -f hono-template.yml | oc create -f -
+
+OpenShift templates allow to use *parameters* which can customize provided
+templates. If you want to specify template parameters from the command line
+use the following syntax:
+
+    oc process -f hono-template.yml \
+      -p "ENMASSE_NAMESPACE=enmasse" \
+      -p "GIT_REPOSITORY=https://github.com/your/hono.git" \
+      -p "GIT_BRANCH=0.6.x"| oc create -f -
+
+{{% note title="Align branches" %}}
+By default the Hono template uses the `master` branch for deploying Hono. As
+this branch might be unstable this may not be what you are looking for, but it
+also is the default branch of the Hono Git repository where you checked out
+the deployment template.
+
+It is recommended that when you execute the guide from an alternate branch
+(e.g. `0.6.x`) that you also pass the same branch as `GIT_BRANCH` to the
+template.
+{{% /note %}}
+
+## Setting up Grafana
+
+Start by creating a new project using:
+
+    oc new-project grafana --display-name='Grafana Dashboard'
+
+Then deploy the Grafana instance using:
+
+    oc process -f ../openshift/grafana-template.yml \
+      -p ADMIN_PASSWORD=admin | oc create -f -
+
+After the Grafana instance is up and running, deploy the default resources:
+
+    ../configure_grafana.sh "$(oc get route grafana -o jsonpath='{.spec.host}')" 80
+
+## Configuring the installation
+
+There are a few ways to configure the deployment. The following sub-sections
+will give you an introduction.
+
+### Configure maximum number of devices per tenant
+
+The default settings for the example Hono device registry limit the number of
+devices to 100 per tenant. If this is not enough for your setup you can change
+the setting by executing the following command, which will increate the number
+to 10.000 devices per tenant:
+
+    oc env dc/hono-service-device-registry HONO_REGISTRY_SVC_MAX_DEVICES_PER_TENANT=10000
+
+## Using the installation
+
+All following examples make use of the running Hono instance. They pretty much
+follow the other examples. Please note that, compared to the more simple
+OpenShift deployment, in this case we need to take care of different project
+names when looking up route host names via `oc get`. You can use the command
+line argument `-n <project>` to specify the project name without changing the
+default selected project.
+
+All examples in the following sub-sections assume that you are located in the
+`examples` directory.
+
+### Extract certificates
+
+In order to connect the external consumer to EnMasse, we need to use a
+proper SSL certificate. We can extract one from the OpenShift using the
+following command:
+
+    oc extract secret/external-certs-messaging --to=target/keys -n enmasse
```

Comment:
target/keys doesn't work by default ... maybe using target/config/hono-demo-certs-jar/ as we do in other document is a better option

Reply:
Well the following commands all have the changed paths. Which makes the commands a bit shorter. But I agree, I you want to re-use the commands from the other tutorials, you might run into issued. I will sync this with the other examples.